### ISSUE TITLE
small bug with gcc-8.1.1

### ISSUE THREAD

**Issue by Pustovetov** [facebook/rocksdb#4017]
Small bug/warning with gcc-8.1.1
```
--------------
util/status.cc: In static member function ‘static const char* rocksdb::Status::CopyState(const char*)’:
util/status.cc:28:15: error: ‘char* strncpy(char*, const char*, size_t)’ output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
   std::strncpy(result, state, cch - 1);
   ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
util/status.cc:19:18: note: length computed here
       std::strlen(state) + 1; // +1 for the null terminator
       ~~~~~~~~~~~^~~~~~~
cc1plus: all warnings being treated as `errors`
------------------------
```
bugfix very simple. 
std::strncpy(result, state, cch);
i.e. we copy '\0' too

**Comment by anand1976** [comment]
Makes sense. Not a bug since we set the null terminator on line 22, but we can do what you suggested to silence the compiler. Do you want to go ahead and submit a PR with this change?

**Comment by Pustovetov** [comment]
Yes, of course it is not real bug. The new gcc just panics. ;)  But if you make such a change that will be very good.

**Comment by ordian** [comment]
I'm having this compilation error when building 5.13.4 with gcc-8.1.1:
```
In file included from memtable/skiplistrep.cc:6:
./memtable/inlineskiplist.h: In instantiation of ‘void rocksdb::InlineSkipList<Comparator>::Node::StashHeight(int) [with Comparator = const rocksdb::MemTableRep::KeyComparator&]’:
./memtable/inlineskiplist.h:612:3:   required from ‘rocksdb::InlineSkipList<Comparator>::Node* rocksdb::InlineSkipList<Comparator>::AllocateNode(size_t, int) [with Comparator = const rocksdb::MemTableRep::KeyComparator&; size_t = long unsigned int]’
./memtable/inlineskiplist.h:574:13:   required from ‘rocksdb::InlineSkipList<Comparator>::InlineSkipList(Comparator, rocksdb::Allocator*, int32_t, int32_t) [with Comparator = const rocksdb::MemTableRep::KeyComparator&; int32_t = int]’
memtable/skiplistrep.cc:28:28:   required from here
./memtable/inlineskiplist.h:283:11: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct std::atomic<rocksdb::InlineSkipList<const rocksdb::MemTableRep::KeyComparator&>::Node*>’ with no trivial copy-assignment [-Werror=class-memaccess]
     memcpy(&next_[0], &height, sizeof(int));
     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ./memtable/inlineskiplist.h:47,
                 from memtable/skiplistrep.cc:6:
/usr/include/c++/8.1.1/atomic:352:12: note: ‘struct std::atomic<rocksdb::InlineSkipList<const rocksdb::MemTableRep::KeyComparator&>::Node*>’ declared here
     struct atomic<_Tp*>
            ^~~~~~~~~~~~
  CC       shared-objects/memtable/vectorrep.o
```
Am I doing something stupid?
`cmake -DCMAKE_BUILD_TYPE=Release -DWITH_JEMALLOC=true -DWITH_SNAPPY=true .. && make`

**Comment by anand1976** [comment]
Closed by #3870
